onnxruntime/core/providers/migraphx/migraphx_call.cc

```diff
@@ -48,8 +48,8 @@ std::conditional_t<THRW, void, Status> RocmCall(
       (void)hipGetDevice(&currentHipDevice);
       (void)hipGetLastError();  // clear last HIP error
       static char str[1024];
-      snprintf(str, 1024, "%s failure %d: %s ; GPU=%d ; hostname=%s ; file=%s ; line=%d ; expr=%s; %s",
-               libName, (int)retCode, RocmErrString(retCode), currentHipDevice,
+      snprintf(str, sizeof(str), "%s failure %d: %s ; GPU=%d ; hostname=%s ; file=%s ; line=%d ; expr=%s; %s",
```

Comment:
This looks like a security bug. Please check the function's document (for Windows and Linux)

Reply:
snprintf does not ensure the dest string is nul-terminated .

Reply:
You didn't introduce the security bug. But, the security bug still exists and it should be fixed. You may open a new PR for  this if you'd like.
On Windows we prefer to use _snprintf , which has a C++ version that can handle such situations well. 
Visual Studio's static analyzer usually can find such issues.

Reply:
I will fix that here.

Reply:
snprintf was replaced with use of std::stringstream. Marking this as resolved.